mapping.go

```diff
@@ -129,9 +146,29 @@ func (o *mapAssignmentKind) String() string {
 	if o.ElemDirect {
 		s += " (direct)"
 	}
+	if o.ElemConvert {
+		s += " (convert)"
+	}
 	return s
 }
 
+func convertibleButNotIdentical(typ, typeDecode types.Type) bool {
+	// Only allow this for basic and named.
+
+	switch typeDecode.(type) {
+	case *types.Basic, *types.Named:
+	default:
+		return false
+	}
+
+	if types.ConvertibleTo(typ, typeDecode) && types.ConvertibleTo(typeDecode, typ) {
```

Comment:
`typeDecode` is the most underlying type for `typ`.

- For pointers this is the type of the ultimate memory location
- For named types, it is the type of the real underlying type

This check only lets this special code execute if BOTH the left and right are convertible to their underlying type (which in effect selects `type LEFT RIGHT` types).
Then we skip in the case of a type that is its own identity.